outlines/prompts.py

```diff
@@ -118,10 +125,14 @@ def _template_from_file(_, path: Path) -> jinja2.Template:
             keep_trailing_newline=True,
             undefined=jinja2.StrictUndefined,
         )
+
+        for name, filter_fn in filters.items():
+            env.filters[name] = filter_fn
+
         return env.get_template(os.path.basename(path))
 
 
-def prompt(fn: Callable) -> Prompt:
+def prompt(fn: Optional[Callable] = None, **filters: Dict[str, Callable]) -> Callable:
```

Comment:
`filters` should be a simple argument that has type `Dict[str, Callable]` and maps the filter names to the functions.

Reply:
I agree, ideally, it would be less surprising for users to have the most similar environment (and behavior) for any method used to create a `Prompt`. Otherwise, it's a great feature addition!